Sources/NIOWebSocket/Base64.swift

```diff
@@ -33,80 +33,66 @@ internal struct Base64 {
   static func encode<Buffer: Collection>(bytes: Buffer)
     -> String where Buffer.Element == UInt8
   {
+    guard bytes.isEmpty == false else {
+        return ""
+    }
     // In Base64, 3 bytes become 4 output characters, and we pad to the
     // nearest multiple of four.
-    let newCapacity = ((bytes.count + 2) / 3) * 4
+    let base64StringLength = ((bytes.count + 2) / 3) * 4
     let alphabet = Base64.encodeBase64
     
-    // NOTE: Once SE-263 lands we should replace this implementation with one
-    //       that makes use of the non copy initializer. For more information
-    //       please see:
-    //       https://github.com/apple/swift-evolution/blob/master/proposals/0263-string-uninitialized-initializer.md
-    var outputBytes = [UInt8]()
-    outputBytes.reserveCapacity(newCapacity)
+    // temporary buffer we use to store the output as zero terminated C string
+    let outputBuffer = UnsafeMutableBufferPointer<UInt8>.allocate(capacity: base64StringLength + 1)
```

Comment:
I’m fairly disinclined to use an `UnsafeMutableBufferPointer` here. However, I would be more inclined to use `String.init(unsafeUninitializedCapacity:initializingWith:)`. We will need to provide a backport of that initializer, but one already exists in `swift-nio-http2` which we can bring over here: https://github.com/apple/swift-nio-http2/blob/1ef8bac96a4bbbc870dcf3c9825df56982188620/Sources/NIOHPACK/HuffmanCoding.swift#L247-L289

Reply:
My benchmarks showed that the cString initialiser is still a tiny bit faster and it does not need a back port.
But I guess, I should introduce a bench mark in this PR first. 
This will make discussing and comparing strategies much better.